app/src/main/java/com/quran/labs/androidquran/ui/helpers/QuranListAdapter.java

```diff
@@ -248,6 +249,35 @@ public boolean onLongClick(View v) {
     return false;
   }
 
+  public void showHideSuraTranslatedName(boolean showSuraTranslatedName) {
+    if (showSuraTranslatedName && elements[1].text.indexOf("(") != -1) return;
+    if (!showSuraTranslatedName && elements[1].text.indexOf("(") == -1) return;
+    for (QuranRow row : elements) {
+      if (row.sura != 0) {
+        if (showSuraTranslatedName) {
+          addTranslatedNameToSuraName(row);
+        } else {
+          removeSuraTranslatedName(row);
+        }
+      }
+    }
+    notifyDataSetChanged();
+  }
+
+  private void removeSuraTranslatedName(QuranRow row) {
+    if (row.text.indexOf("(") != -1)
+      row.text = row.text.substring(0, row.text.indexOf("(") - 1);
+  }
+
+  private void addTranslatedNameToSuraName(QuranRow row) {
+    StringBuilder builder = new StringBuilder(row.text);
+    String translation = context.getResources().getStringArray(R.array.sura_names_translation)[row.sura - 1];
+    if (!TextUtils.isEmpty(translation)) {
+      builder.append(" (" + translation + ")");
+    }
+    row.text = builder.toString();
+  }
+
```

Comment:
suggestion - instead of doing it this way, i would remove this code and let changing the variable cause `SuraListFragment` instead to recompute with the updated flag and `notifyDataSetChanged`.

Reply:
so you could replace that code with a check in `onResume` - if the value of the checkbox changed, then re-compute and `notifyDataSetChanged`, otherwise do nothing.

Reply:
okay, I will move this code [three methods] to the SuraListFragment, but I track if the value changed or not using the below-stopping conditions before re-compute. 

 if (showSuraTranslatedName && elements[1].text.indexOf("(") != -1) return;
 if (!showSuraTranslatedName && elements[1].text.indexOf("(") == -1) return;

- do you know how to track the value after being changed ?

Reply:
no need to move this code - remove it altogether. in `onCreate` of the `SuraListFragment`, read the value of `wantTranslations` into a variable. in `onResume`, check if the actual value differs, and if so, just re-call `getSuraList()` - pass the updated value to the adapter (you might have to expose a setter).